Generals/Code/GameEngine/Source/Common/System/Debug.cpp

```diff
@@ -148,11 +148,14 @@ static void doStackDump();
 // ----------------------------------------------------------------------------
 inline Bool ignoringAsserts()
 {
+	Bool ignore = !DX8Wrapper_IsWindowed;
 #ifdef DEBUG_CRASHING
-	return !DX8Wrapper_IsWindowed || (TheGlobalData&&TheGlobalData->m_debugIgnoreAsserts);
-#else
-	return !DX8Wrapper_IsWindowed;
+	if (TheGlobalData && TheGlobalData->m_debugIgnoreAsserts)
+		ignore = true;
 #endif
+	if (TheGlobalData && TheGlobalData->m_headless)
+		ignore = true;
+	return ignore;
```

Comment:
I think this can be further simplified:

```cpp
inline Bool ignoringAsserts()
{
	if (!DX8Wrapper_IsWindowed)
		return true;
	if (TheGlobalData && TheGlobalData->m_headless)
		return true;
#ifdef DEBUG_CRASHING
	if (TheGlobalData && TheGlobalData->m_debugIgnoreAsserts)
		return true;
#endif

	return false;
}